pkg/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -129,6 +130,12 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 	}
 
 	for _, hc := range allMachineHealthChecks.Items {
+		if hasRemediationDisabledAnnotation(*machine) {
```

Comment:
please move this check outside of loop, because we do not really need to check all MHC's